R/subset_nhdplus.R

```diff
@@ -782,74 +790,105 @@ subset_vpu <- function(fline, vpu,
 #' sample_flines <- readRDS(staged_nhdplus$flowline)
 #'
 #' subset_rpu(sample_flines, rpu = "07b")
-subset_rpu <- function(fline, rpu, run_make_standalone = TRUE) {
-  fline <- check_names(fline, "subset_rpu")
-
-  # Find all outlets of current rpu and sort by size
-  # !ToNode %in% FromNode finds non-terminal flowlines that exit the domain.
-  outlets <- filter(drop_geometry(fline), .data$RPUID %in% rpu)
-
-  if("tocomid" %in% names(outlets)) outlets <- dplyr::rename(outlets, toCOMID = .data$tocomid)
-
-  if(any(c("tocomid", "toCOMID") %in% names(outlets))) {
-
-    outlets <- filter(outlets, .data$Hydroseq == .data$TerminalPa |
-                        !.data$toCOMID %in% .data$COMID)
-  } else {
+subset_rpu <- function(fline, rpu, run_make_standalone = TRUE, strict = FALSE) {
+  orig_names <- names(fline)
```

Comment:
This is essentially a rewrite. The original logic was left intact but the strict flag was added. See code comments for clarification of what's going on.

Reply:
Changed to lowercase in code throughout while I was in here.